File: src/main/java/com/profe_me_quiero_graduar/app/service/mapper/AnunciosMapper.java
package com.profe_me_quiero_graduar.app.service.mapper;

import com.profe_me_quiero_graduar.app.domain.*;
import com.profe_me_quiero_graduar.app.service.dto.AnunciosDTO;
import org.mapstruct.*;

/**
 * Mapper for the entity {@link Anuncios} and its DTO {@link AnunciosDTO}.
 */
@Mapper(componentModel = "spring", uses = {})
public interface AnunciosMapper extends EntityMapper<AnunciosDTO, Anuncios> {
    @Named("id")
    @BeanMapping(ignoreByDefault = true)
    @Mapping(target = "id", source = "id")
    AnunciosDTO toDtoId(Anuncios anuncios);
}

File: server/src/service/mapper/anuncios.mapper.ts
import { Anuncios } from '../../domain/anuncios.entity';
import { AnunciosDTO } from '../dto/anuncios.dto';

/**
 * A Anuncios mapper object.
 */
export class AnunciosMapper {
    static fromDTOtoEntity(entityDTO: AnunciosDTO): Anuncios {
        if (!entityDTO) {
            return;
        }
        let entity = new Anuncios();
        const fields = Object.getOwnPropertyNames(entityDTO);
        fields.forEach((field) => {
            entity[field] = entityDTO[field];
        });
        return entity;
    }

    static fromEntityToDTO(entity: Anuncios): AnunciosDTO {
        if (!entity) {
            return;
        }
        let entityDTO = new AnunciosDTO();

        const fields = Object.getOwnPropertyNames(entity);

        fields.forEach((field) => {
            entityDTO[field] = entity[field];
        });

        return entityDTO;
    }
}
